### ISSUE TITLE
Daily extracts are more than one day old

### ISSUE THREAD

**Issue by aapris** [osmlab/osm-qa-tiles#28]
It seems that planet and country extracts are today over 3 weeks old.

E.g. "DOWNLOAD PLANET (24.3 GB) **updated daily**" and finland are dated:

```
$ curl -I https://s3.amazonaws.com/mapbox/osm-qa-tiles-production/latest.planet.mbtiles.gz 2>&1 | grep Last-Modified
Last-Modified: Fri, 03 Apr 2020 22:44:42 GMT
$ curl -I https://s3.amazonaws.com/mapbox/osm-qa-tiles-production/latest.country/finland.mbtiles.gz 2>&1 | grep Last-Modified
Last-Modified: Fri, 03 Apr 2020 22:26:17 GMT
```

Is there an error in extract process?

**Comment by mikelmaron** [comment]
Thanks for raising the issue here @aapris. We (well @sbma44 and @mattficke) have been aware that builds are failing, and have been unable to identify the root cause and correct the issues; though likely related to growing size and particulars in our platform. We haven't been able to dedicate as much time as we'd like, as Mapbox is not longer using OSM QA Tiles internally. 

We've been considering some time how to transition OSM QA Tiles generation to elsewhere, to someone or some group who are actively consuming the tiles and would be invested in their continuing upkeep and operation, such as HOT (/cc @bopercival-hot) who run https://osm-analytics.org/#/ or @jenningsanderson. We are working to make the generation code open (it's a relatively short script) and can advise on what it would take to get running elsewhere.

So that's the state of things. Would really love to see OSM QA Tiles on solid feet soon!

**Comment by aapris** [comment]
We are actively mapping some OSM details (entrances) in Helsinki/Finland and we desperately need most recent qa-tiles of this area. Perhaps we could try to build them by ourselves from finland-latest.osm.pbf? We know how to extract tiles from **mbtiles** file, but do you have build instructions for it?

E.g. https://github.com/mapbox/mbtiles-spec/wiki/Implementations lists so many so many options that I am confused.

**Comment by sbma44** [comment]
@aapris using a smaller extract as your source may simplify the problem. The core of the script is extracting the contents of a `.osm.pbf` to geojson, then turning that geojson into an `mbtiles` file with [tippecanoe](https://github.com/mapbox/tippecanoe). Here's an example invocation:

```
osmium export  \ 
     -c "osm-qa-tile.osmiumconfig" --overwrite \ 
     -f geojsonseq -r -o - \ 
     --verbose "input.osm.pbf" |  \
     tippecanoe -q -l osm -n osm-latest -o "output.mbtiles\" -f -z12 -Z12 -ps -pf -pk -P -b0 -d20
```

where osm-qa-tile.osmiumconfig is:

```js
{
  "attributes": {
    "type": true,
    "id": true,
    "version": true,
    "changeset": true,
    "timestamp": true,
    "uid": true,
    "user": true,
    "way_nodes": false
  },
  "linear_tags": [
    "highway",
    "barrier",
    "natural=coastline"
  ],
  "area_tags": [
    "aeroway",
    "amenity",
    "building!=no",
    "landuse",
    "leisure",
    "man_made",
    "natural!=coastline"
  ],
  "exclude_tags": [
    "created_by",
    "source",
    "source:*"
  ],
  "include_tags": []
}
```

**Comment by aapris** [comment]
Thank you so much! This seems to work (although I have to check it with our GIS expert).

The command is this (for the records):

`osmium export -c osm-qa-tile.osmiumconfig --overwrite -f geojsonseq -r -o - --verbose 2020-04-29/finland-latest.osm.pbf | tippecanoe -q -l osm -n osm-latest -o "finland-latest.mbtiles" -f -z12 -Z12 -ps -pf -pk -P -b0 -d20`

And extracting to qa-tiles:

`mb-util --image_format=pbf finland-latest.mbtiles /path/to/osm-qa-tiles`

And before I could compile tippecanoe I had to 
`apt install osmium-tool libsqlite3-dev`

Oh, and in nginx config we have location block like this:

```
    location /osm-qa-tiles {
        add_header 'Access-Control-Allow-Origin' '*';
        add_header Content-Encoding gzip;
        alias /path/to/osm-qa-tiles;
    }
```

And if you are interested at all why we are doing this, we are trying to improve last 50 meters routing in citylogistics. Here is the demo app:
https://app.gatesolve.com/route/60.173483271344,24.953021684712155/60.17360292053621,24.95075439237752/

**Comment by tyrasd** [comment]
> advise on what it would take to get running elsewhere.

Can you tell us more about hardware requirements? I don't know for sure, but if requirements are not too high, we @GIScience / HeiGIT might be able to help, maybe.

**Comment by aapris** [comment]
It takes about 15 minutes to convert `finland-latest.osm.pbf` to `finland-latest.mbtiles` (osmium export | tippecanoe) on Hetzner's cloud server CX31 (2 vCPU cores, 8 GB RAM).

**Comment by tyrasd** [comment]
Ok. But I guess processing the full planet osm-qa-tiles would be quite a bit more challenging. Our main use case for the osm-qa-tiles is for https://osm-analytics.org and requires planet data. :bow:

**Comment by aapris** [comment]
I'm just trying with europe.pbf (22 GB). It seems osmium makes very effective use of multiple CPU cores (I've now 8), CPU usage (according to `top`) is around 800% now. :) It seems that spinning hard disks are my bottleneck, have an SSD. It also seems that osmium+tippecanoe is using >30 GB memory now (`top`'s RES column), so I think 64 GB RAM is a good idea.

I guess this will take some hours to complete and I'm afraid I've to go offline soon, so this time I can't get more results about this.

**Comment by sbma44** [comment]
@aapris @tyrasd you've hit on the crux of why open sourcing this hasn't been quick: prior to the regressions in performance, we were generating the planet mbtiles in a container with a reservation size of >100GB of RAM and 16 cores (in EC2 terms, this works out to about an r5.4xlarge, which costs about USD $1/hr as I write this). For architectural reasons not worth going into we were optimizing to avoid disk use as much as possible, which also contributed toward a resource-hungry but fast run. With this config the run took about 9 hours (I believe this includes the subsequent step to split out per-country extracts, which could be done with fewer resources).

My hope had been to adjust the script to use more disk and less RAM, moving it into the realm of running on spare hardware. This has not gone as smoothly as I'd hoped, but I remain optimistic that with 16GB and a fast SSD, acceptable performance will be achievable.

TLDR: if you need a planet.mbtiles right this minute, I think that spending $10 on an r5.4xlarge has a good chance of delivering it.

**Comment by jenningsanderson** [comment]
Thanks @sbma44 for all the work you're doing to keep qa-tiles alive and the insights on this thread!

@aapris: If you're specifically focused on highways/sidewalks/etc, consider adding an intermediate filtering step with [osmium tags-filter](https://docs.osmcode.org/osmium/latest/osmium-tags-filter.html). In this way, your workflow would be:

finland.osm.pbf > `osmium tags-filter` > finland-highways.osm.pbf > `osmium export` > finland-highways.geojsonseq > `tippecanoe` > finland-highways.mbtiles

It doesn't get much faster than `osmium`, so doing this processing up front is best. If you're not using large polygons / buildings / etc. in your analysis, this will save considerable time and resources. 

Additionally, since you're focusing on smaller regions, you might be able to use larger-scale tiles (zoom 15, for example) to get more fine-grained units of analysis (at the tile level). Your `tippecanoe` flags would then be: `-Pf -z15 -Z15 -ps -pf -pk -b0 -d17`. qa-tiles at z12 are phenomenal for planet-scale analysis, but for smaller regions, z15 offers much finer resolution!

@tyrasd, what do you think about a regular release of object-specific osm-qa-tiles in addition to the `planet-latest.mbtiles` file?  For example, `planet-highways.mbtiles` and `planet-buildings.mbtiles` tilesets?  These would be made using the workflow mentioned above.  Or perhaps `planet-roads-highways-pois.mbtiles` containing all roads/highways/pois. For reference, the last time I tried this, it's a difference of nearly 50GB in file size from latest-planet.mbtiles to an object-specific tileset.  This wouldn't be a replacement of latest.planet.mbtiles, but could be helpful for some specific use cases.

**Comment by tuukka** [comment]
@jenningsanderson Thank you for the guidance - the current tile size is an issue to us indeed! The "layer" approach is also intriguing and would add a lot of flexibility as I could decide which layers I needed to fetch with a big impact on resource use.

**Comment by jenningsanderson** [comment]
@tuukka and @aapris - the [`osmium`](https://github.com/osmlab/osm-qa-tiles/tree/osmium) branch of this repository also contains these instructions, for your reference. I just updated them. I am also building a full planet osm-qa-tiles file right now that I will share when finished.

**Comment by jenningsanderson** [comment]
If these are helpful, I'll share them here. Generated last week with the planet file as of May 18, 2020. 

See the [README](https://github.com/osmlab/osm-qa-tiles/blob/osmium/README.md) for more information on how these were generated: 

osm-qa-tiles (equivalent to the updated version of the standard osm-qa-tiles) 
[osm-qa-tiles.mbtiles](https://openstreetmap-analysis.s3-us-west-1.amazonaws.com/osm-qa-tiles.mbtiles)

Without large objects and coastlines (filtered): 
[osm-qa-tiles-compact.mbtiles](https://openstreetmap-analysis.s3-us-west-1.amazonaws.com/osm-qa-tiles-compact.mbtiles)

Compact, and with turn restriction representations! 
[osm-qa-with-tr.mbtiles](https://openstreetmap-analysis.s3-us-west-1.amazonaws.com/osm-qa-with-tr.mbtiles)

**Comment by mikelmaron** [comment]
HOT is now producing weekly tiles
